Interface.C

```diff
@@ -134,28 +150,91 @@ void preciceAdapter::Interface::configureMesh(const fvMesh& mesh)
             }
         }
 
+
         // Pass the mesh vertices information to preCICE
-        precice_.setMeshVertices(meshID_, numDataLocations_, vertices, vertexIDs_);
-    }
-    else
-    {
-        FatalErrorInFunction
-             << "ERROR: interface points location type "
-             << locationsType_
-             << " is invalid."
-             << exit(FatalError);
+        precice_.setMeshVertices(NodemeshID_, numNodeLocations_, vertices, NodeIDs_);
+
+
+        //Only set the triangles, when it is necessary
+        if (locationsType_ == "faceTriangles"){
+            {
+
+                for (uint j = 0; j < patchIDs_.size(); j++)
+                {
+
+                    //Define triangles
+
+                    const List<face> faceField=mesh.boundaryMesh()[patchIDs_.at(j)].localFaces();
+
+                    const Field<point> pointCoords= mesh.boundaryMesh()[patchIDs_.at(j)].localPoints();
+
+                    //Array to transform coords in precice IDs
+                    double triaCoords[faceField.size()*18];
+
+                    unsigned int coordIndex=0;
+
+                    forAll(faceField,facei){
+
+                        const face& quad_face=faceField[facei];
+
+                        faceTriangulation fT(pointCoords,quad_face,false);
+
+                        for(uint triaIndex=0; triaIndex<2; triaIndex++){
+                            for(uint nodeIndex=0; nodeIndex<3; nodeIndex++){
+                                for(uint xyz=0; xyz<3; xyz++)
```

Comment:
Nice index name! I think OpenFOAM uses `dim`, though. But you can keep it like this, I like it! ;-)

Reply:
Ok then I will keep it.